### PR TITLE
A11y: Change links to buttons

### PR DESCRIPTION
**Pull request by patrickhlauke**
- more semantically accurate, will expose these as buttons to assistive technologies (in general, whenever you have a `href="#"` that's a strong indication that it's actually a `<button>` you want to use)
- add `type="button"` to all buttons that lack type - just to be safe
- note: while having block-level elements like `<div>` inside a `<button>` is an html validation error, it does not affect functionality as long as there's no more structure inside it

### CODE REVIEW

**Comment by patrickhlauke** [comment]
changing to draft for now, as i've noticed some instances where the default button styles now override some of the styles that were previously taken from/applied to the links the extension (wrongly) used instead. will tweak this further to hopefully catch these instances.

would be good to gauge interest in this PR though before I spend considerably more time on it. i.e. are there non-obvious reasons why links, rather than buttons, have been used for these controls?

**Comment by eliykat** [comment]
Hi @patrickhlauke, I've noticed your issues and PRs relating to accessibility and I really appreciate you spending time in this area. I have your PRs on my to-do list and hopefully won't keep you waiting too long, we're just quite busy at this point in the release cycle.

If `<button>` elements can be made to look and behave the same as `<a>` elements do currently, but are more accessible and more semantically accurate, then that sounds like a win to me. That said, I'll double check with the team to make sure there's nothing I'm missing.

**Comment by patrickhlauke** [comment]
hey @eliykat, good to hear. sorry, my message wasn't intended as a "i submitted this X days ago, why has it not been actioned/merged yet", but more of a general gauge of how likely the idea was to be considered. sorry, re-reading the above i see my tone was leaning far too much towards giving the impression it was the former. long story short: all good, no meaning to rush you folks. i'll fiddle some more with the styles and once i get to the point where i think i caught all stylings/made them look the same as now, i'll set the PR back to ready for review 👍

File: src/popup/accounts/home.component.html [hunk 1]
```diff
@@ -1,14 +1,14 @@
...
-</div>
-<a routerLink="/environment" class="settings-icon">
-    <i class="fa fa-cog fa-lg" aria-hidden="true"></i><span>&nbsp;{{'settings' | i18n}}</span>
-</a>
+<div class="center-content">
+    <div class="content">
+        <div class="logo-image"></div>
+        <p class="lead text-center">{{'loginOrCreateNewAccount' | i18n}}</p>
+        <a class="btn primary block" routerLink="/login"><b>{{'login' | i18n}}</b></a>
+        <button (click)="launchSsoBrowser()" class="btn block">
```
Comment (addisonbeck) [inline src/popup/accounts/home.component.html:6]:
```suggestion
        <button type="button" (click)="launchSsoBrowser()" class="btn block">

```

File: src/popup/accounts/lock.component.html [hunk 2]
```diff
@@ -38,11 +38,11 @@
         </div>
         <div class="box" *ngIf="biometricLock">
             <div class="box-footer">
-                <button class="btn primary block" (click)="unlockBiometric()" appStopClick>{{'unlockWithBiometrics' | i18n}}</button>
+                <button type="button" class="btn primary block" (click)="unlockBiometric()" appStopClick>{{'unlockWithBiometrics' | i18n}}</button>
             </div>
         </div>
         <p class="text-center">
-            <a href="#" appStopClick (click)="logOut()">{{'logOut' | i18n}}</a>
+            <button appStopClick (click)="logOut()">{{'logOut' | i18n}}</button>
```
Comment (addisonbeck) [inline src/popup/accounts/lock.component.html:45]:
```suggestion
            <button type="button" appStopClick (click)="logOut()">{{'logOut' | i18n}}</button>
```

**Review by addisonbeck — COMMENTED** [review]
Hi @patrickhlauke , just wanted to let you know getting this merged is on my immediate todo list.

>  sorry, re-reading the above i see my tone was leaning far too much towards giving the impression it was the former

I didn't read anything forward in your tone at all, no worries. This PR is certainly something we are interested in. There is no non-obvious underlying reason `<a>` tags are used instead of buttons here, other than (as you discovered) that there is preexisting styling that gets in the way here.

Otherwise, I noticed a couple places that were missing `type="button"`, listed as suggestions below.

**Comment by patrickhlauke** [comment]
spent some more time on this PR, and i *think* i've now got everything now looking exactly as it previously did (i.e. changing the links to buttons left no discernible visual/layout differences) - unless I overlooked something somewhere

**Comment by patrickhlauke** [comment]
If possible, suggest merging this soon, as otherwise it'll keep hitting merge conflicts (spent some time tonight just fixing things up)

File: src/popup/scss/buttons.scss [hunk 3]
```diff
@@ -1,88 +1,101 @@
...
+    display: block;
+    width: 100%;
+    text-align: left;
+}
+
+button {
+    border: none;
+    background: transparent;
+    color: inherit;
+}
```
Comment (addisonbeck) [inline src/popup/scss/buttons.scss:101]: I'm not sure what's going on with this diff, but it is showing the entire file as changed. Possibly the same line break issue as your other [PR](https://github.com/bitwarden/browser/pull/1990#issuecomment-930971418)?

File: src/popup/accounts/home.component.html [hunk 4]
```diff
@@ -1,14 +1,15 @@
...
-<a routerLink="/environment" class="settings-icon">
-    <i class="fa fa-cog fa-lg" aria-hidden="true"></i><span>&nbsp;{{'settings' | i18n}}</span>
-</a>
+<div class="center-content">
+    <div class="content">
+        <div class="logo-image"></div>
+        <p class="lead text-center">{{'loginOrCreateNewAccount' | i18n}}</p>
+        <a class="btn primary block" routerLink="/login"><b>{{'login' | i18n}}</b></a>
+        <button type="button" (click)="launchSsoBrowser()" class="btn block">
+
```
Comment (addisonbeck) [inline src/popup/accounts/home.component.html:7]:
```suggestion
```

**Review by addisonbeck — CHANGES_REQUESTED** [review]
I think we are seeing unexpected line break control changes on this one too, and one other quick formatting fix. After that we should be able to get this merged in.

**Comment by patrickhlauke** [comment]
sorry, mea culpa yes ... serves me right for not configuring VS Code properly on the second machine. will push up a commit to fix this

**Comment by patrickhlauke** [comment]
Line ending encoding fixed

**Review by addisonbeck — APPROVED** [review]
Major thanks again @patrickhlauke 🙏

**Comment by patrickhlauke** [comment]
no worries @addisonbeck ... my pleasure (these are the low-hanging fruit changes ... there's a few more fundamental things that will be more tricky to tackle, but we'll get to them bit by bit ;) )

**Pull request merged**